pyaction/pyaction_wconprod_insert_kw.py

```diff
@@ -0,0 +1,206 @@
+
+    FGPR = summary_state["FGPR"]
+    print("FGPR {}\n".format(FGPR))
+
+    if (WGOR_OP01 >= FU_GOR and FU_GASFL == 0 and current_time.month >= 1 and current_time.year >= 2019):
+        print("PYACTION version of ACT-01 triggered at {}\n".format(current_time))
+        kw = """
+        WCONPROD
+        OP01     OPEN   ORAT WU_WOPR  1*     1*    1*    1*      150   1*    5      1* /
+        /"""
+        schedule.insert_keywords(kw, report_step)
+    if (WGOR_OP02 >= 150  and FU_GASFL == 0 and current_time.month >= 1 and current_time.year >= 2019):
+        print("PYACTION version of ACT-02 triggered at {}\n".format(current_time))
+        kw = """
+        WCONPROD
+        OP02     OPEN   ORAT WU_WOPR  1*     1*    1*    1*      150   1*    5      1* /
+        /"""
+        schedule.insert_keywords(kw, report_step)
+    if (WGOR_OP03 >= 150  and FU_GASFL == 0 and current_time.month >= 1 and current_time.year >= 2019):
+        print("PYACTION version of ACT-03 triggered at {}\n".format(current_time))
+        kw = """
+        WCONPROD
+        OP03     OPEN   ORAT WU_WOPR  1*     1*    1*    1*      150   1*    5      1* /
+        /"""
+        schedule.insert_keywords(kw, report_step)
+# The group property field is somehow not available, when this ACTION-04 is not an ACTION-X..
+#    if (FU_GASFL == 0 and GOPR_FIELD <= 500):
+#        print("PYACTION version of ACT-04 triggered at {}\n".format(current_time))
+#        summary_state["FU_GASFL"] = 1 # Gas Field Flag Set to On
+#        kw = """
+#        GCONPROD
+#        FIELD    GRAT  1*     1*     3.0E6  1*     1*     1*    1*     1*     1*       /
+#        /
+#        WCONPROD
+#        OP01     OPEN   GRUP   1*     1*     1*    1*     1*     50.0   20   5         /
+#        OP02     OPEN   GRUP   1*     1*     1*    1*     1*     50.0   20   5         /
+#        OP03     OPEN   GRUP   1*     1*     1*    1*     1*     50.0   20   5         /
+#        /"""
+#        schedule.insert_keywords(kw, report_step)
+    if (FU_GASFL == 1 and FGPR <= 2.0E6):
+        print("PYACTION version of ACT-05 triggered at {}\n".format(current_time))
+        kw = """
+        WCONPROD
+        'OP*'    SHUT   1*     1*     1*     1*    1*     1*     1*                    /
+        /
+        WCONINJE
+        'GI*'    GAS    SHUT   GRUP  1*     1*     1*     1*    1*                     /
+        'WI*'    WAT    SHUT   GRUP  1*     1*     1*     1*    1*                     /
+        /"""
+        schedule.insert_keywords(kw, report_step)
+
+#-- ---------------------------------------------------------------------------------------------------------------------------------
+#-- ACTIONX WCONPROD - BEAM BACK OIL RATE IF GOR >= 150 M3/M3 FOR WELL OP01
+#-- ---------------------------------------------------------------------------------------------------------------------------------
+#ACTIONX
+#        ACT-01    500  10                                  /
```

Comment:
I think the last two parameters might not be fully represented in your script and datafile, yet. We might need multiple PYACTION keywords and python scripts to accomplish this.

From the manual:
ACTNSTEP is a positive integer that defines the number times that the
ACTNAME definition is executed. ACTIONX definitions are activated at
the end of a time step and this parameter is used to set how many time
steps the ACTNAME definition will be invoked. The default value of one
means that the definition will be executed only once. Use a large value, for
example 10,000 for the definition to be executed at every time step.
Noted that the counter only affects successful evaluations; i.e. if
ACTNSTEP is set equal to one (the default), then the simulator will test
the action at the end of every time step until it evaluates to true.

ACTDELTA is a real positive value that stipulates the duration of time that
the conditions defined on the second record to be satisfied before the
ACTIONX action are executed. For example, if ACTDELTA is defaulted
the actions will be executed at the end of the time step for which the
conditions are met. If set to say 30, then a minimum of 30 days will pass
before the actions are executed (assuming field or metric units).